src/pages/dataElements/form/fields.tsx

```diff
@@ -24,10 +28,122 @@ import {
 } from '../../../lib'
 import classes from './customFields.module.css'
 import { EditableFieldWrapper } from './EditableFieldWrapper'
+import { useIsFieldValueUnique } from './useIsFieldValueUnique'
+
+export function NameField() {
+    const params = useParams()
+    const dataElementId = params.id as string
+    const checkIsValueTaken = useIsFieldValueUnique({
+        field: 'name',
+        id: dataElementId,
+    })
+    const { meta } = useField('name', {
+        subscription: { validating: true },
+    })
+
+    return (
+        <FieldRFF
+            loading={meta.validating}
+            component={InputFieldFF}
+            dataTest="dataelementsformfields-name"
+            required
+            inputWidth="400px"
+            label={i18n.t('{{fieldLabel}} (required)', {
+                fieldLabel: i18n.t('Name'),
+            })}
+            name="name"
+            helpText={i18n.t(
+                'A data element name should be concise and easy to recognize.'
+            )}
+            // Do not pass checkIsValueTaken directly, otherwise memoization
+            // won't work due to additional arguments being passed
+            validate={(name: string) => checkIsValueTaken(name)}
+            validateFields={[]}
+        />
+    )
+}
+
+export function ShortNameField() {
+    const params = useParams()
+    const dataElementId = params.id as string
+    const checkIsValueTaken = useIsFieldValueUnique({
+        field: 'shortName',
+        id: dataElementId,
+    })
+    const { meta } = useField('shortName', {
+        subscription: { validating: true },
+    })
+
+    return (
+        <FieldRFF
```

Comment:
I think it could make sense to have a small wrapper around `FieldRFF` that could simplify some things. Not sure if it's worth the change right now though.

It could generate `dataTest` dynamically, based on `name-prop`.
Pass `validateFields={[]}` by default (eliminate need to add the prop for every field - it's easy to forget). 
Could calculate `required` based on zod.schema. 
Could automatically add `uniqueValidation` based on a prop. 

I think would further simplify adding new fields. It does have the downside of some more logic - but it would be easier to make changes to every field.

Reply:
> Not sure if it's worth the change right now though.

I would prefer to not change it now and see how things develop. We can always add a concretion for that later on.

---

> Some fields have field-validation for required, is that needed?

Right now not needed